MULE-18691: Add documentation informing that Parameter Groups must implement equals and hashcode. (#76)
mulesoft/docs-mule-sdk: modules/ROOT/pages/define-parameters.adoc

image::mule-sdk-dsl-parameter-pojos.png[Decision tree for POJOs and parameter groups]

> * Implementing *equals* and *hashCode* methods

Reviewer: ```suggestion
* Implementing `equals` and `hashCode` methods
```

> Java classes used for POJO parameters or Parameter Groups *must* implement the *equals* and

Reviewer: ```suggestion
Java classes used for POJO parameters or Parameter Groups must implement the `equals` and
```

> POJO parameters (regardless if they are used with the @ParameterGroup annotation) *must* override the *equals* and

Reviewer 2: I would say that the classes are the ones that must override the method instead of the parameters and the parameter groups.

> POJO parameters (regardless if they are used with the @ParameterGroup annotation) *must* override the *equals* and

Reviewer 2: I would say implement instead of override.

> *hashCode* methods for <<static-dynamic-configs.adoc#Dynamic-Configuration-Lifecycle,Dynamic Configurations>> to function correctly.

Reviewer: ```suggestion
`hashCode` methods for <<static-dynamic-configs.adoc#Dynamic-Configuration-Lifecycle,Dynamic Configurations>> to function correctly.
```

> Even if no configurations have POJO parameters, it is still required that they implement *equals* and *hashCode* as future

Reviewer: ```suggestion
Even if existing configurations do not have POJO parameters, the configurations need `equals` and `hashCode` methods because future
```

> runtime functionality may depend on it. Below is an example from a class in the Database Connector that illustrates how to do this:

Reviewer: ```suggestion
runtime functionality may depend on it. The following example from a class in the Database Connector that illustrates the implementation of `equals` and `hashcode` methods:
```
